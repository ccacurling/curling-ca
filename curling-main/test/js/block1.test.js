import React from 'react';
import { Posts } from "./block1";
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('Posts component', () => {
	/** Mock WordPress post to test with**/
	const posts = [
		{
			title: {
				rendered: 'Hi Roy'
			},
			content: {
				rendered: 'Lorem ipsum, etc.'
			}
		},
		{
			title: {
				rendered: 'Hi Mike'
			},
			content: {
				rendered: 'Lorem ipsum, etc.'
			}
		},
	];


	it('Matches snapshot with basic props', () => {
		const component = renderer.create(
			<Posts
				posts={posts}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

	it('Applies className to outermost element', () => {
		const component = renderer.create(
			<Posts
				posts={posts}
				className={'food'}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});
});
