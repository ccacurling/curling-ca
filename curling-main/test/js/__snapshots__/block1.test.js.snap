// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts component Applies className to outermost element 1`] = `
<div
  className="food"
>
  <p>
    Hi Roys
  </p>
</div>
`;

exports[`Posts component Matches snapshot with basic props 1`] = `
<div
  className="post-list-wrapper"
>
  <p>
    Hi Roys
  </p>
</div>
`;
